src/scanner.h

```diff
@@ -104,4 +105,33 @@ typedef struct ScannerCtx
 extern int ts_scanner_scan(ScannerCtx *ctx);
 extern bool ts_scanner_scan_one(ScannerCtx *ctx, bool fail_if_not_found, char *item_type);
 
+typedef struct InternalScannerCtx InternalScannerCtx;
+typedef struct ScanIterator
+{
+	ScannerCtx ctx;
```

Comment:
Wondering if there's now any good reason to distinguish between `ScannerCtx` and `ScanIterator`? I mean, is there any place where we are not using `ScanIterator` or could we just have expanded `ScannerCtx` with the additional fields in `ScanIterator` (and possibly rename)?

Just asking because the additional indirection of struct embedding here seems potentially unnecessary.

Reply:
So now that we have an embedded scankey in the Iterator I think it's a bit odd to have the scanIterator be the same thing as the ScannerCtx (for example, the Ctx would then have both a pointer to a scankey and an embedded scankey). For backwards-compatibility I think we should keep the ScannerCtx as is but just add the iterator. I can mock-up the one-object version if you disagree though